client/client_pool/src/concord_client_pool.cpp

```diff
@@ -361,6 +379,18 @@ void ConcordClientPool::CreatePool(concord::config_pool::ConcordClientPoolConfig
   jobs_queue_max_size_ = config.external_requests_queue_size;
 }
 
+void ConcordClientPool::AddSenderAndSignature(std::vector<uint8_t> &request, const ClientPtr &chosenClient) {
+  concord::messages::ReconfigurationRequest rreq;
+  concord::messages::deserialize(request, rreq);
+  rreq.sender = static_cast<decltype(rreq.sender)>(chosenClient->getClientId());
```

Comment:
The sender-id is a part of the ClientRequestMsgHeader, why do you need it inside the request, too?

Reply:
In

[concord-bft/reconfiguration/cmf/concord.cmf](https://github.com/vmware/concord-bft/blob/0bb1834b467342c5f449a3508975c2cc05ca8d28/reconfiguration/cmf/concord.cmf#L419)

Line 419 in [0bb1834](https://github.com/vmware/concord-bft/commit/0bb1834b467342c5f449a3508975c2cc05ca8d28)

 Msg ReconfigurationRequest 1 { 

We can see the following declaration (snipped for easy reference)
Msg ReconfigurationRequest 1 {
uint64 sender
bytes signature
oneof {
\<many reconfiguration commands\>
} command
bytes additional_data
}

Reconfiguration request contains an additional sender id field. So we had to add signature and sender at that point when client_pool picks the actual client (external client --> bft client)